src/components/Api.js

```diff
@@ -0,0 +1,146 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          return Promise.reject(`Unexpected Error: ${res.status}`);
+        }
+      })
+      .catch((err) => {
+        console.error(err); // log the error to the console
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`   after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.